### PR TITLE
Add GetAllCustomPropertyValues for repositories

### PR DESCRIPTION
**Pull request by liaodaniel**
Closes https://github.com/google/go-github/issues/3014.

Signed-off-by: Daniel Liao <10663736+liaodaniel@users.noreply.github.com>

### CODE REVIEW

File: github/repos_properties.go
Comment (gmlewis) [inline github/repos_properties.go:1]:
```suggestion
// Copyright 2023 The go-github AUTHORS. All rights reserved.
//
// Use of this source code is governed by a BSD-style
// license that can be found in the LICENSE file.

package github
```

File: github/repos_properties.go
Comment (gmlewis) [inline github/repos_properties.go:8]:
```suggestion
// GetAllCustomPropertyValues gets all custom property values that are set for a repository.
```

File: github/repos_properties.go
Comment (gmlewis) [inline github/repos_properties.go:14]:
```suggestion
	u := fmt.Sprintf("repos/%v/%v/properties/values", org, repo)
```

File: github/repos_properties.go
Comment (gmlewis) [inline github/repos_properties.go:13]:
```suggestion
func (s *RepositoriesService) GetAllCustomPropertyValues(ctx context.Context, org, repo string) ([]*CustomPropertyValue, *Response, error) {
```

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @liaodaniel !
Just a few tweaks, please, then we will be ready for a second LGTM+Approval from any other contributor to this repo before merging.

**Comment by liaodaniel** [comment]
Thank you @gmlewis, need to pay more attention to copying and pasting 😅

**Review by gmlewis — APPROVED** [review]
Thank you, @liaodaniel !
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

**Review by valbeat — APPROVED** [review]
LGTM!

**Comment by gmlewis** [comment]
Thank you, @valbeat !
Merging.

**Pull request merged**
